File: app/release/app-release/sources/com/bilibili/bililive/batteria/imageloader/ImageLoaderActivity.java
package com.bilibili.bililive.batteria.imageloader;

import a.g;
import a.w.c.i;
import android.content.Context;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import com.bilibili.bililive.tests.R;
import d.b.k.h;
import e.b.a.a.f.c;
import e.b.a.a.f.d.b;
import java.io.File;

@g(d1 = {"\u0000(\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\u0018\u00002\u00020\u0001B\u0005¢\u0006\u0002\u0010\u0002J\n\u0010\u0003\u001a\u0004\u0018\u00010\u0004H\u0002J\b\u0010\u0005\u001a\u00020\u0006H\u0002J\u0012\u0010\u0007\u001a\u00020\u00062\b\u0010\b\u001a\u0004\u0018\u00010\tH\u0014J\u0010\u0010\n\u001a\u00020\u00062\u0006\u0010\u000b\u001a\u00020\fH\u0002¨\u0006\r"}, d2 = {"Lcom/bilibili/bililive/batteria/imageloader/ImageLoaderActivity;", "Landroidx/appcompat/app/AppCompatActivity;", "()V", "getLiveModManagerCacheDir", "Ljava/io/File;", "hookObject", "", "onCreate", "savedInstanceState", "Landroid/os/Bundle;", "startAnimation", "image", "Landroid/widget/ImageView;", "app_release"}, mv = {1, 1, 16})
public final class ImageLoaderActivity extends h {

    public static final class a implements View.OnClickListener {

        /* renamed from: c  reason: collision with root package name */
        public final /* synthetic */ ImageLoaderActivity f2596c;

        public a(ImageLoaderActivity imageLoaderActivity) {
            this.f2596c = imageLoaderActivity;
        }

        public final void onClick(View view) {
            File a2 = ImageLoaderActivity.a(this.f2596c);
            if (a2 != null) {
                a2.listFiles();
            }
            System.out.println();
        }
    }

    public static final /* synthetic */ File a(ImageLoaderActivity imageLoaderActivity) {
        File filesDir;
        File parentFile;
        Context applicationContext = imageLoaderActivity.getApplicationContext();
        String path = (applicationContext == null || (filesDir = applicationContext.getFilesDir()) == null || (parentFile = filesDir.getParentFile()) == null) ? null : parentFile.getPath();
        try {
            return new File(path + "/app_mod_resource/cache/live");
        } catch (Exception e2) {
            e2.printStackTrace();
            return null;
        }
    }

    @Override // androidx.activity.ComponentActivity, d.b.k.h, d.g.d.b, d.j.d.d
    public void onCreate(Bundle bundle) {
        super.onCreate(bundle);
        setContentView(R.layout.activity_imageloader);
        ImageView imageView = (ImageView) findViewById(R.id.image1);
        c cVar = new c();
        cVar.f3751c = new b(this);
        i.a((Object) imageView, "imageView1");
        imageView.setTag("http://i0.hdslb.com/bfs/live/user_cover/2a246a601047fbbe524af3849ce48cb7a170065d.jpg@420w_236h_1e_1c_85q.webp");
        cVar.f3750a.submit(new e.b.a.a.f.a(cVar, "http://i0.hdslb.com/bfs/live/user_cover/2a246a601047fbbe524af3849ce48cb7a170065d.jpg@420w_236h_1e_1c_85q.webp", imageView));
        imageView.setOnClickListener(new a(this));
    }
}

File: app/src/main/java/com/bilibili/bililive/tests/imageloader/ImageLoaderActivity.kt
package com.bilibili.bililive.tests.imageloader

import android.animation.AnimatorSet
import android.animation.ObjectAnimator
import android.os.Bundle
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import com.bilibili.bililive.tests.R
import com.bilibili.bililive.tests.imageloader.cache.DoubleCache
import java.io.File

/**
 * @author: yaobeihaoyu
 * @version: 1.0
 * @since: 2020/7/2 3:48 PM
 * @description:
 */
class ImageLoaderActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_imageloader)

        val imageView1 = findViewById<ImageView>(R.id.image1)

        val imageLoader = ImageLoader()
        imageLoader.cache = DoubleCache(this)
        imageLoader.displayImage(imageView1, "http://i0.hdslb.com/bfs/live/user_cover/2a246a601047fbbe524af3849ce48cb7a170065d.jpg@420w_236h_1e_1c_85q.webp")

        imageView1.setOnClickListener {
//            startAnimation(imageView1)
            val file = getLiveModManagerCacheDir()
            val list = file?.listFiles()
            println()
        }
    }

    private fun startAnimation(image: ImageView) {
        val scaleXAnim = ObjectAnimator.ofFloat(image, "scaleX", 1.3f, 1f)
        val scaleYAnim = ObjectAnimator.ofFloat(image, "scaleY", 1.3f, 1f)
        val alphaAnim = ObjectAnimator.ofFloat(image, "alpha", 0f, 1f)
        val animSet = AnimatorSet()
        animSet.playTogether(scaleXAnim, scaleYAnim, alphaAnim)
        animSet.duration = 300
        animSet.start()
    }

    private fun hookObject() {
        val clazz = Class.forName("com.bilibili.bililive.tests.test.TestSingleton")
        val field = clazz.getField("INSTANCE")
        val a = field.get(null)
        val method = clazz.getDeclaredMethod("test")
        method.invoke(a)
    }

    private fun getLiveModManagerCacheDir(): File? {
        val appPath = applicationContext?.filesDir?.parentFile?.path
        try {
            return File("$appPath/app_mod_resource/cache/live")
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return null
    }
}